packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -385,40 +370,124 @@ function setProp(
-          }
+        const propertyInfo = getPropertyInfo(key);
+        if (propertyInfo !== null) {
+          setValueForProperty(domElement, propertyInfo, value);
         } else {
-          if (
-            // shouldIgnoreAttribute
-            // We have already filtered out reserved words.
-            key.length > 2 &&
-            (key[0] === 'o' || key[0] === 'O') &&
-            (key[1] === 'n' || key[1] === 'N')
-          ) {
-            return;
-          }
+          setValueForAttribute(domElement, key, value);
+        }
+      }
+    }
+  }
+}
 
-          const propertyInfo = getPropertyInfo(key);
-          if (propertyInfo !== null) {
-            setValueForProperty(domElement, propertyInfo, value);
+function setPropOnCustomElement(
+  domElement: Element,
+  tag: string,
+  key: string,
+  value: mixed,
+  props: any,
+): void {
+  switch (key) {
+    case 'style': {
+      // Relies on `updateStylesByID` not mutating `styleUpdates`.
+      setValueForStyles(domElement, value);
+      break;
+    }
+    case 'dangerouslySetInnerHTML': {
+      if (value != null) {
+        if (typeof value !== 'object' || !('__html' in value)) {
+          throw new Error(
+            '`props.dangerouslySetInnerHTML` must be in the form `{__html: ...}`. ' +
+              'Please visit https://reactjs.org/link/dangerously-set-inner-html ' +
+              'for more information.',
+          );
+        }
+        const nextHtml: any = value.__html;
+        if (nextHtml != null) {
+          if (props.children != null) {
+            throw new Error(
+              'Can only set one of `children` or `props.dangerouslySetInnerHTML`.',
+            );
+          }
+          if (disableIEWorkarounds) {
+            domElement.innerHTML = nextHtml;
           } else {
-            setValueForAttribute(domElement, key, value);
+            setInnerHTML(domElement, nextHtml);
           }
         }
       }
+      break;
+    }
+    case 'children': {
+      if (typeof value === 'string') {
+        setTextContent(domElement, value);
+      } else if (typeof value === 'number') {
+        setTextContent(domElement, '' + value);
+      }
+      break;
+    }
+    case 'onScroll': {
+      if (value != null) {
+        if (__DEV__ && typeof value !== 'function') {
+          warnForInvalidEventListener(key, value);
+        }
+        listenToNonDelegatedEvent('scroll', domElement);
+      }
+      break;
+    }
+    case 'onClick': {
+      // TODO: This cast may not be sound for SVG, MathML or custom elements.
+      if (value != null) {
+        if (__DEV__ && typeof value !== 'function') {
+          warnForInvalidEventListener(key, value);
+        }
+        trapClickOnNonInteractiveElement(((domElement: any): HTMLElement));
+      }
+      break;
+    }
+    case 'suppressContentEditableWarning':
+    case 'suppressHydrationWarning':
+    case 'innerHTML': {
+      // Noop
+      break;
+    }
```

Comment:
is no longer skipping `defaultValue defaultChecked autoFocus` and no longer setting `multiple muted` technically a breaking change?

Reply:
Yea but probably fine. Since this path is no longer taken for `is` etc where this might be relevant in the weird overlap between `input` and `is`. Seems unlikely people rely on defaultValue as an attribute. As a property is not officially shipped yet so not breaking.

`autoFocus` is interesting because I noticed that we only allow it on specific elements, and not for example something with a tabIndex on it which seems like an existing React bug. So actually setting autoFocus can at least give you a chance to handle that in the custom element.

`multiple` and `muted` will be reinstated to optionally use properties with the `enableCustomElementPropertySupport` flag on.